src/Parsers/ParserInsertQuery.cpp

```diff
@@ -256,14 +256,23 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     if (infile)
     {
         query->infile = infile;
+        query->compression = compression;
+
+        if (infile)
+            query->children.push_back(infile);
         if (compression)
-            query->compression = compression;
+            query->children.push_back(compression);
     }
 
     if (table_function)
     {
         query->table_function = table_function;
         query->partition_by = partition_by_expr;
+
+        if (table_function)
+            query->children.push_back(table_function);
```

Comment:
Here was a problem. `ParserInsertQuery` didn't add `table_function` as its child and thus [IAST::childrenHaveSecretParts()](https://github.com/ClickHouse/ClickHouse/blob/0dcb98565b471f2c5fc715c3498d5a4d5a311e04/src/Parsers/IAST.cpp#L185) didn't understand when an INSERT INTO TABLE statement has secret parts.